administration/src/bp-modules/stores/StoresCSVInput.tsx

```diff
@@ -45,22 +57,66 @@ const hasStoreDuplicates = (stores: AcceptingStoreEntry[]) => {
   return (
     new Set(
       stores.map(({ data }: { data: StoreData }) =>
-        JSON.stringify([data['name'], data['street'], data['houseNumber'], data['postalCode'], data['location']])
+        JSON.stringify([
+          data[FIELD_NAME],
+          data[FIELD_STREET],
+          data[FIELD_HOUSE_NUMBER],
+          data[FIELD_POSTAL_CODE],
+          data[FIELD_LOCATION],
+        ])
       )
     ).size < stores.length
   )
 }
 
-const StoresCsvInput = ({ setAcceptingStores, fields }: StoresCsvInputProps): ReactElement => {
-  const [inputState, setInputState] = useState<FileInputStateType>('idle')
+const getStoreCoordinates = (
```

Comment:
@ztefanie 
I splitted this function and moved it to a separate file.
But i really dunno how i can get rid of this differing return type.
Since i would like to keep this in a `Promise.all` to update the state once all requests have been made. If i split this up and use separate fetches its getting more messy.

Maybe you can give me short suggestion how you would implement this.

Reply:
Thanks for splitting this up, you can add a Promise.resolve manually, in the first if-case.
`if (store.hasValidCoordinates()) return Promise.resolve(store)`